hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsNode.java

```diff
@@ -210,12 +209,17 @@ protected boolean enoughResources(Offer offer, double cpus, int mem) {
   private List<Task> createTasks(Offer offer) {
     String executorName = getExecutorName();
     String taskIdName = String.format("%s.%s.%d", name, executorName, System.currentTimeMillis());
-    List<Task> tasks = new ArrayList<Task>();
+    List<Task> tasks = new ArrayList<>();
 
+    String nnNum = null;
     for (String type : getTaskTypes()) {
-      List<Resource> resources = getTaskResources(type);
-      ExecutorInfo execInfo = createExecutor(taskIdName, name, executorName);
       String taskName = getNextTaskName(type);
+      if (nnNum == null && taskName.startsWith("namenode")) {
```

Comment:
why would there be more than 1 namenode task though? When launching a list of tasks, there is only one NN. When we launch a list of tasks, I believe we launch NN, and ZKFC together but not two NN's.